Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -490,6 +491,152 @@ ThrowCompletionOr<Optional<String>> to_smallest_temporal_unit(GlobalObject& glob
+            // i. Return value.
+            return value;
+        }
+
+        // b. If value has an [[InitializedTemporalDate]] internal slot, then
+        if (is<PlainDate>(value_object)) {
+            auto& plain_date_object = static_cast<PlainDate&>(value_object);
+
+            // i. Return ? CreateTemporalDateTime(value.[[ISOYear]], value.[[ISOMonth]], value.[[ISODay]], 0, 0, 0, 0, 0, 0, value.[[Calendar]]).
+            return TRY(create_temporal_date_time(global_object, plain_date_object.iso_year(), plain_date_object.iso_month(), plain_date_object.iso_day(), 0, 0, 0, 0, 0, 0, plain_date_object.calendar()));
+        }
+
+        // c. Let calendar be ? GetTemporalCalendarWithISODefault(value).
+        calendar = TRY(get_temporal_calendar_with_iso_default(global_object, value_object));
+
+        // d. Let fieldNames be ? CalendarFields(calendar, « "day", "hour", "microsecond", "millisecond", "minute", "month", "monthCode", "nanosecond", "second", "year" »).
+        auto field_names = TRY(calendar_fields(global_object, *calendar, { "day"sv, "hour"sv, "microsecond"sv, "millisecond"sv, "minute"sv, "month"sv, "monthCode"sv, "nanosecond"sv, "second"sv, "year"sv }));
+
+        // e. Let fields be ? PrepareTemporalFields(value, fieldNames, «»).
+        auto* fields = TRY(prepare_temporal_fields(global_object, value_object, field_names, {}));
+
+        // f. Let dateOptions be ! OrdinaryObjectCreate(null).
+        auto* date_options = Object::create(global_object, nullptr);
+
+        // g. Perform ! CreateDataPropertyOrThrow(dateOptions, "overflow", "constrain").
+        MUST(date_options->create_data_property_or_throw(vm.names.overflow, js_string(vm, "constrain"sv)));
+
+        // h. Let result be ? InterpretTemporalDateTimeFields(calendar, fields, dateOptions).
+        result = TRY(interpret_temporal_date_time_fields(global_object, *calendar, *fields, *date_options));
+
+        // i. Let offsetString be ? Get(value, "offset").
+        offset_string = TRY(value_object.get(vm.names.offset));
+
+        // j. Let timeZone be ? Get(value, "timeZone").
+        time_zone = TRY(value_object.get(vm.names.timeZone));
+
+        // k. If offsetString is undefined, then
+        if (offset_string.is_undefined()) {
+            // i. Set offsetBehaviour to wall.
+            offset_behavior = OffsetBehavior::Wall;
+        }
+    }
+    // 7. Else,
+    else {
+        // a. Let string be ? ToString(value).
+        auto string = TRY(value.to_string(global_object));
+
+        // b. Let result be ? ParseTemporalRelativeToString(string).
+        auto parsed_result = TRY(parse_temporal_relative_to_string(global_object, string));
+
+        // NOTE: The ISODateTime struct inside parsed_result will be moved into `result` at the end of this path to avoid mismatching names.
+        //       Thus, all remaining references to `result` in this path actually refers to `parsed_result`.
```

Comment:
```suggestion
        // NOTE: The ISODateTime struct inside `parsed_result` will be moved into `result` at the end of this path to avoid mismatching names.
        //       Thus, all remaining references to `result` in this path actually refer to `parsed_result`.
```